File: exercism/python/rna-transcription/rna_transcription.py
conversions = {
  "C" : "G",
  "G" : "C",
  "T" : "A",
  "A" : "U"
}

def to_rna(dna_strand):
  rna = ""
  for x in dna_strand:
    if conversions.setdefault(x, None) != None:
      rna += conversions[x]
    else:
      return None
  return rna

File: exercism/javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {};

DnaTranscriber.prototype.toRna = function(Dna) {
  if (RegExp("[^CGAT]").exec(Dna)) {
    throw Error("Invalid input");
  } else {
    var translator = {
      C: "G",
      G: "C",
      A: "U",
      T: "A"
    };
    var Rna = "";
    for (var i = 0; i < Dna.length; i++) {
      Rna += translator[Dna[i]];
    }
    return Rna;
  }
};

module.exports = DnaTranscriber;